packages/block-library/src/table-of-contents/hooks.js

```diff
@@ -97,6 +98,11 @@ function getLatestHeadings( select, clientId ) {
 			if ( blockName === 'core/heading' ) {
 				const headingAttributes = getBlockAttributes( blockClientId );
 
+				// Skip headings that are deeper than maxLevel
+				if ( maxLevel !== null && headingAttributes.level > maxLevel ) {
+					continue;
+				}
```

Comment:
```suggestion
				// Skip headings that are deeper than maxLevel
				if ( maxLevel && headingAttributes.level > maxLevel ) {
					continue;
				}
```

It will avoid false positives on super rare occasions if `maxLevel` is undefined, the `getBlockAttributes` returns `undefined|null`, and the fallback is used.

Reply:
Thanks! That's a good idea. I've updated my patch accordingly.